GNUmakefile

```diff
@@ -46,66 +66,135 @@ endif
+$(warning VERSION not set! Verify setuptools_scm[toml] is installed and try again.)
+endif
 
-.PHONY: $(phonies)
+#############################################################################
+# Post Release handling
+# "post" releases are useful for when non-code changes are made after
+# a release was created:
+# - When a README is updated, or CHANGE_LOG after a release was made
+# - When build changes occur for distributing the application to another
+#	platform but the code has zero changes
+#############################################################################
+
+# NOTE: 1.0.0 and 1.0.0.post0 mean the same thing, so to keep things simple if a post0 stable tag is detected it
+#		will be truncated.
+export VERSION := $(shell echo $(VERSION) | sed -E 's/\.post0$$.*//')
+
+ifneq (,$(findstring post, $(VERSION)))
+
+	export RELEASE := $(shell echo $(VERSION) | sed -En 's/.*post([1-9]+)$$/\1/p')
+
+	# The RPM version starts at 1, whereas the Python post version starts at 0 (e.g. RPM 1.0.0-1 == Py 1.0.0.post0).
+	# Add 1 to translate the Python post version to RPM.
+#	ifeq ($(RELEASE),)
+#	export RELEASE=1
+#	else
+	export RELEASE := $(shell expr $(RELEASE) + '1')
+#	endif
+
+	# If the version is A.B.C.postN (with no other suffix), then bump the RELEASE number in the RPM and trim the suffix on the VERSION.
+	# Otherwise if there is a suffix after postN, it should be preserved. When a suffix exists after postN, that means a
+	# development branch is being used and the version should be preserved to indicate that context. When there is NO suffix after
+	# postN, that means this is a re-release (a repackaging) of an already published version.
+	# e.g.
+	# 1.7.1.post1      translates to RPM speak as 1.7.1-2 (the post release preserves the same version but indicates the re-packaging).
+	# 1.7.1.post2.dev0 translates to RPM speak as 1.7.1.post2.dev0-1 (the entire version remains untouched)
+	# See
+	export VERSION := $(shell echo $(VERSION) | sed -E 's/\.post[1-9]+$$//')
+
+else
+	# Always set the RELEASE to 1 to indicate this build is the first release to be published for the version.
+	export RELEASE=1
+endif
 
-# Quick run shellspec to do a one off test
-test:
-	$(SHELLSPEC)
+# After the VERSION has been normalized, make the image version.
+# Image versions should never have the Python post version included if they're stable, it's confusing to image users.
+# Image users simply pull the image and fetch the new layers, whereas RPM users have to pragmatically know when an RPM
+# is newer (e.g. YUM/Zypper/apt needs a way to convey the repackaging).
+# - Undo the RPM tilde, sanitize the version; image tags do not like tildes and are okay with dashes, unlike RPMs
+# - Replace any '+' with '_' because image tags don't like the '+' character
+ifeq ($(IMAGE_VERSION),)
+export IMAGE_VERSION := $(shell echo $(VERSION) | tr -s '~' '-' | tr -s '+' '_' | sed 's/^v//')
+endif
 
-# Run tests against all the shell intepreters in SHELLS
-test-all:
```

Comment:
I (ab)use test/test-all a lot why are we removing them?

Reply:
I'm surprised this is in here, I didn't mean to remove these.